kucher/view/main_window/register_view_widget/import_export_dialog.py

```diff
@@ -0,0 +1,302 @@
+    """
+    dialog_box = QWidget()
+    dialog_box.setGeometry(10, 10, 1000, 700)
+    file_name, _ = QFileDialog.getOpenFileName(dialog_box, 'Import configuration file', '',
+                                               'YAML Files (*.yml *.yaml);;All Files (*)')
+    if not file_name:
+        return
+    try:
+        with open(file_name, 'r') as file:
+            imported_registers = yaml.load(file, Loader=yaml.Loader)
+            file.close()
+    except IOError as ex:
+        _logger.exception(f'File {file_name} could not be open')
+        show_error('Import failed',
+                   f'Cannot open {file_name}',
+                   f'Error: {str(ex)}',
+                   parent)
+        return
+
+    except Exception as ex:
+        file.close()
+        _logger.exception(f'File {file_name} could not be parsed')
+        show_error('Import failed',
+                   f'Cannot read {file_name}',
+                   f'Error: {str(ex)}',
+                   parent)
+        return
+
+    if imported_registers:
+        result, detail = check_registers(registers, imported_registers)
+        if result == CheckResult.NO_ERROR:
+            write_registers(parent, file_name, registers, imported_registers)
+        else:
+            show_error_box(result, detail, file_name, parent)
+
+
+def check_registers(registers, imported_registers):
+    try:
+        for reg_check in registers:
+            if reg_check.name in imported_registers:
+                if not (reg_check.mutable and reg_check.persistent):
+                    _logger.error(f'Import failed: this parameter cannot be modified on this device: {reg_check}')
+                    return CheckResult.NOT_MUTABLE, reg_check.name
+
+                elif not check_type(reg_check, imported_registers[reg_check.name]):
+                    _logger.error(f'Import failed: this parameter type is incorrect {reg_check}')
+                    return CheckResult.INCORRECT_TYPE, reg_check.name
+
+                elif len(imported_registers[reg_check.name]) != len(reg_check.cached_value):
+                    _logger.error(f'Import failed: this parameter dimension is incorrect {reg_check}')
+                    return CheckResult.INCORRECT_DIMENSION, reg_check.name
+
+                elif reg_check.has_min_and_max_values and reg_check.type_id != Register.ValueType.BOOLEAN:
+                    if not (reg_check.min_value <= imported_registers[reg_check.name] <= reg_check.max_value):
+                        _logger.error(f'Import failed: this parameter value is outside permitted range {reg_check}')
+                        return CheckResult.OUTSIDE_RANGE, reg_check.name
+
+    except Exception as ex:
+        _logger.exception(f'Could not write registers: {str(ex)}')
+        return CheckResult.UNKNOWN, str(ex)
+
+    return CheckResult.NO_ERROR, ''
+
+
+def show_error_box(result, detail, file_name, parent):
```

Comment:
Type annotations would help. Not just here but in all functions that you've defined.